js/packages/openinference-instrumentation-llama-index/src/utils.ts

```diff
@@ -56,14 +85,73 @@ function getExecContext(span: Span) {
   return execContext;
 }
 
-export function patchQueryMethod(
-  original: typeof module.RetrieverQueryEngine.prototype.query,
-  module: typeof llamaindex,
+/**
+ * If execution results in an error, push the error to the span.
+ * @param span
+ * @param error
+ */
+function handleError(span: Span, error: Error | undefined) {
+  if (error) {
+    span.recordException(error);
+    span.setStatus({
+      code: SpanStatusCode.ERROR,
+      message: error.message,
+    });
+    span.end();
+  }
+}
```

Comment:
I think you can leave this as a function in the instrumentation file

Reply:
This is only used for the methods in utils?